modules/common/topology/topology.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2025 Red Hat
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package topology
+
+import (
+	"context"
+	"fmt"
+	topologyv1 "github.com/openstack-k8s-operators/infra-operator/apis/topology/v1beta1"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/util"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// EnsureTopologyRef - retrieve the Topology CR referenced and add a finalizer
+func EnsureTopologyRef(
+	ctx context.Context,
+	h *helper.Helper,
+	topologyRef *TopologyRef,
+	finalizer string,
+) (*topologyv1.Topology, string, error) {
+
+	var err error
+	var hash string
+
+	// no Topology is passed at all or it is missing some data
```

Comment:
@stuggi @olliewalsh I understand you might don't like the compact format of the `if` statement on L43.
It is equivalent to:
```golang
if topologyRef == nil {
    return nil, "", fmt.Errorf("No valid TopologyRef input passed")
}
if topologyRef.Name == "" || topologyRef.Namespace == "" {
   return nil, "", fmt.Errorf("No valid TopologyRef input passed")
}
```
which might result easier to follow in terms of readability. Let me know which form you prefer, as I'm more inclined to keep the compact one.

Reply:
I have no preference